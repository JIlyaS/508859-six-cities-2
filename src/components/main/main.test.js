import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main';

it(`Main correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Main
      rentList={[]}
      onDetailModalClick={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
