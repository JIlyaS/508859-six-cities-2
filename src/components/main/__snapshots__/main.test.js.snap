// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main correctly renders after relaunch 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <Connect(CityList) />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            4
             places to stay in 
            Amsterdam
          </b>
          <WithSortList />
          <Connect(OfferList)
            offers={
              Array [
                Object {
                  "city": Object {
                    "location": Object {
                      "latitude": 52.3909553943508,
                      "longitude": 4.85309666406198,
                      "zoom": 12,
                    },
                    "name": "Amsterdam",
                  },
                  "coordinate": Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  "features": Array [
                    "Entire place",
                    "3 Bedrooms",
                    "Max 4 adults",
                  ],
                  "hostUser": Object {
                    "avatar": "avatar-angelina.jpg",
                    "description": Array [
                      "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                      "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                    ],
                    "name": "Angelina",
                    "status": "pro",
                  },
                  "id": "id0",
                  "img": "apartment-01.jpg",
                  "insideProperties": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                    "Heating",
                    "Coffee machine",
                    "Baby seat",
                    "Kitchen",
                    "Dishwasher",
                    "Cabel TV",
                    "Fridge",
                  ],
                  "isPremium": true,
                  "photos": Array [
                    "room.jpg",
                    "apartment-01.jpg",
                    "apartment-02.jpg",
                    "apartment-03.jpg",
                    "studio-01.jpg",
                    "apartment-01.jpg",
                  ],
                  "price": 110,
                  "rating": 4.8,
                  "reviews": Array [
                    Object {
                      "avatar": "avatar-max.jpg",
                      "date": "April 2019",
                      "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                      "id": "review1",
                      "name": "Max",
                      "rating": 4.7,
                    },
                    Object {
                      "avatar": "avatar-max.jpg",
                      "date": "May 2019",
                      "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                      "id": "review2",
                      "name": "Alex",
                      "rating": 4,
                    },
                  ],
                  "title": "Beautiful & luxurious apartment at great location",
                  "type": "apartment",
                },
                Object {
                  "city": Object {
                    "location": Object {
                      "latitude": 52.3909553943508,
                      "longitude": 4.85309666406198,
                      "zoom": 12,
                    },
                    "name": "Amsterdam",
                  },
                  "coordinate": Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  "features": Array [
                    "Entire place",
                    "3 Bedrooms",
                    "Max 4 adults",
                  ],
                  "hostUser": Object {
                    "avatar": "avatar-angelina.jpg",
                    "description": Array [
                      "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                      "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                    ],
                    "name": "Angelina",
                    "status": "pro",
                  },
                  "id": "id1",
                  "img": "room.jpg",
                  "insideProperties": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                    "Heating",
                    "Coffee machine",
                    "Baby seat",
                    "Kitchen",
                    "Dishwasher",
                    "Cabel TV",
                    "Fridge",
                  ],
                  "isPremium": false,
                  "photos": Array [
                    "room.jpg",
                    "apartment-01.jpg",
                    "apartment-02.jpg",
                    "apartment-03.jpg",
                    "studio-01.jpg",
                    "apartment-01.jpg",
                  ],
                  "price": 120,
                  "rating": 4,
                  "reviews": Array [
                    Object {
                      "avatar": "avatar-max.jpg",
                      "date": "April 2019",
                      "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                      "id": "review1",
                      "name": "Max",
                      "rating": 4.7,
                    },
                  ],
                  "title": "Wood and stone place",
                  "type": "private room",
                },
                Object {
                  "city": Object {
                    "location": Object {
                      "latitude": 52.3909553943508,
                      "longitude": 4.85309666406198,
                      "zoom": 12,
                    },
                    "name": "Amsterdam",
                  },
                  "coordinate": Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  "features": Array [
                    "Entire place",
                    "3 Bedrooms",
                    "Max 4 adults",
                  ],
                  "hostUser": Object {
                    "avatar": "avatar-angelina.jpg",
                    "description": Array [
                      "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                      "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                    ],
                    "name": "Angelina",
                    "status": "pro",
                  },
                  "id": "id2",
                  "img": "apartment-02.jpg",
                  "insideProperties": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                    "Heating",
                    "Coffee machine",
                    "Baby seat",
                    "Kitchen",
                    "Dishwasher",
                    "Cabel TV",
                    "Fridge",
                  ],
                  "isPremium": false,
                  "photos": Array [
                    "room.jpg",
                    "apartment-01.jpg",
                    "apartment-02.jpg",
                    "apartment-03.jpg",
                    "studio-01.jpg",
                    "apartment-01.jpg",
                  ],
                  "price": 200,
                  "rating": 4,
                  "reviews": Array [
                    Object {
                      "avatar": "avatar-max.jpg",
                      "date": "April 2019",
                      "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                      "id": "review1",
                      "name": "Max",
                      "rating": 4.7,
                    },
                  ],
                  "title": "Canal View Prinsengracht",
                  "type": "apartment",
                },
                Object {
                  "city": Object {
                    "location": Object {
                      "latitude": 52.3909553943508,
                      "longitude": 4.85309666406198,
                      "zoom": 12,
                    },
                    "name": "Amsterdam",
                  },
                  "coordinate": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                  "features": Array [
                    "Entire place",
                    "3 Bedrooms",
                    "Max 4 adults",
                  ],
                  "hostUser": Object {
                    "avatar": "avatar-angelina.jpg",
                    "description": Array [
                      "A quiet cozy and picturesque that hides behind a a river
      by the unique lightness of Amsterdam. The building is green
      and from 18th century.",
                      "An independent House, strategically
      located between Rembrand Square and National Opera, but where
      the bustle of the city comes to rest in this alley flowery and
      colorful.",
                    ],
                    "name": "Angelina",
                    "status": "pro",
                  },
                  "id": "id3",
                  "img": "apartment-03.jpg",
                  "insideProperties": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                    "Heating",
                    "Coffee machine",
                    "Baby seat",
                    "Kitchen",
                    "Dishwasher",
                    "Cabel TV",
                    "Fridge",
                  ],
                  "isPremium": true,
                  "photos": Array [
                    "room.jpg",
                    "apartment-01.jpg",
                    "apartment-02.jpg",
                    "apartment-03.jpg",
                    "studio-01.jpg",
                    "apartment-01.jpg",
                  ],
                  "price": 180,
                  "rating": 5,
                  "reviews": Array [
                    Object {
                      "avatar": "avatar-max.jpg",
                      "date": "April 2019",
                      "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                      "id": "review1",
                      "name": "Max",
                      "rating": 4.7,
                    },
                  ],
                  "title": "Nice, cozy, warm big bed apartment",
                  "type": "apartment",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              activeCoordinate={
                Array [
                  52.3909553943508,
                  4.85309666406198,
                ]
              }
              coordinates={
                Array [
                  Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
